code/modules/overmap/ftl_shunt/core.dm

```diff
@@ -0,0 +1,344 @@
+				shake_camera(H, 2, 1)
+
+			if(SHUNT_SEVERITY_MAJOR)
+				to_chat(H, SPAN_WARNING("You feel your insides twisted inside and out as you are violently shunted between dimensions, and you feel like something is watching you!"))
+				if(prob(25))
+					H.hallucination(50, 50)
+				if(prob(15))
+					H.vomit()
+				shake_camera(H, 2, 1)
+
+			if(SHUNT_SEVERITY_CRITICAL)
+				to_chat(H, SPAN_DANGER("You feel an overwhelming sense of nausea and vertigo wash over you, your instincts screaming that something is wrong!"))
+				if(prob(50))
+					H.hallucination(100, 100)
+				if(prob(45))
+					H.vomit()
+				shake_camera(H, 5, 4)
+
+	for(var/obj/machinery/light/L in world) //Fuck with and or break lights.
+		if(!L.z in ftl_computer.linked.map_z)
+			continue
+		switch(shunt_sev)
+			if(SHUNT_SEVERITY_MINOR)
+				if(prob(15))
+					L.flicker()
+			if(SHUNT_SEVERITY_MAJOR)
+				if(prob(35))
+					L.flicker()
+
+	for(var/obj/machinery/power/apc/A in world)
+		if(!A.z in ftl_computer.linked.map_z)
+			continue
+		switch(shunt_sev)
+			if(SHUNT_SEVERITY_MAJOR)
+				if(prob(15))
+					A.energy_fail(rand(30, 80))
+				if(prob(10))
+					A.overload_lighting(25)
+
+			if(SHUNT_SEVERITY_CRITICAL)
+				if(prob(35))
+					A.energy_fail(rand(60, 150))
+				if(prob(50))
+					A.overload_lighting(50)
+
+/obj/machinery/ftl_shunt/core/proc/get_status()
+	if(stat & (BROKEN|NOPOWER))
+		return FTL_STATUS_OFFLINE
+	if(cooldown)
+		return FTL_STATUS_COOLDOWN
+	else
+		return FTL_STATUS_GOOD
+
+/obj/machinery/ftl_shunt/core/proc/get_fuel(var/list/input)
+	var/total_fuel_joules
+
+	for(var/obj/machinery/ftl_shunt/fuel_port/F in input)
+		total_fuel_joules += F.get_avaliable_fuel_joules()
+
+	return total_fuel_joules
+
+/obj/machinery/ftl_shunt/core/proc/get_fuel_maximum(var/list/input)
+	var/max_fuel_joules
+	for(var/obj/machinery/ftl_shunt/fuel_port/F in input)
+		max_fuel_joules += F.get_max_fuel()
+
+	return max_fuel_joules
+
+obj/machinery/ftl_shunt/core/proc/fuelpercentage()
+	if(!fuel_ports.len)
+		return 0
+	return round(100.0*get_fuel(fuel_ports)/get_fuel_maximum(fuel_ports), 0.1)
+
+
+/obj/machinery/ftl_shunt/core/proc/use_fuel(var/joules_req)
+	var/joules_per_port
+	var/avail_fuel = get_fuel(fuel_ports)
+	var/list/fueled_ports = list()
+	var/ports_used
+
+	if(joules_req > avail_fuel) //Not enough fuel in the system.
+		return FALSE
+
+	for(var/obj/machinery/ftl_shunt/fuel_port/F in fuel_ports)
+		if(F.has_fuel())
+			fueled_ports += F
+
+	joules_per_port = (joules_req / fueled_ports.len)
+
+	for(var/obj/machinery/ftl_shunt/fuel_port/F in fueled_ports)
+		if(F.use_fuel_joules(joules_per_port))
+			ports_used++
+
+	if(ports_used == fueled_ports.len)
+		return TRUE
+
+/obj/machinery/ftl_shunt/core/proc/get_charge_time()
+	return (ftl_computer.linked.vessel_mass * CHARGE_TIME_PER_TON) * time_multiplier
+
+/obj/machinery/ftl_shunt/core/Process()
+	if(stat & (BROKEN|NOPOWER) && charging)
```

Comment:
(X & y)

Reply:
?

Reply:
`if((stat & (BROKEN|NOPOWER)) && charging)`
Similar to the issue with 'in', sometimes it can be ambiguous if that will resolve to `(x & y) && z` or x & (y && z)` .